src/app/components/ImageCard/ImageCard.tsx

```diff
@@ -0,0 +1,65 @@
+import Image from 'next/image';
+import { type CSSProperties, memo } from 'react';
+import styles from './ImageCard.module.scss';
+import classNames from 'classnames';
+
+interface ImageCardProps {
+  imageUrl: string;
+  imageAlt: string;
+  children: React.ReactNode;
+  imageFirst?: boolean;
+  imageWidth?: number;
+  imageHeight?: number;
+  imageStyle?: CSSProperties;
+}
+
+const baseClasses = {
+  container:
+    'grid grid-cols-2 justify-items-center gap-4 xl:justify-items-start xl:gap-14',
+  imageBox: 'image-box h-full w-[30rem] overflow-hidden',
+  textBox: 'text-box order-2 flex flex-col gap-5 pb-10 pt-10',
+};
+
+export const ImageCard = memo(function ImageCard({
+  imageUrl,
+  imageAlt,
+  children,
+  imageFirst = true,
+  imageWidth = 0,
+  imageHeight = 0,
+  imageStyle = {},
+}: ImageCardProps) {
+  const containerClasses = classNames(
+    styles['image-card'],
+    baseClasses.container,
+    imageFirst ? 'xl:grid-cols-[1fr_2fr]' : 'xl:grid-cols-[2fr_1fr]',
+  );
```

Comment:
We'll be able to remove these combined class variables if we go the component route mentioned in my other comment 👍